### PR TITLE
feat: created the Author/PR State Icon component

### LINKED ISSUE #3156: Feature: Unified Author/PR State Icon for Pull Requests Table
As a user of the pull requests (PRs) table feature (#3146), I want a unified icon that indicates who the author of the pull request is/was and what the state of the PR is, i.e. Draft, Open, Closed, Merged so that I can have more room to view other information about the pull request in table.

image

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Adds the Author/PR State icon to the project.

## Related Tickets & Documents

Closes #3156

## Mobile & Desktop Screenshots/Recordings

Zoomed in to give a better idea.

CleanShot 2024-04-10 at 11 03 48

## Steps to QA

Check the different states out in Storybook, https://deploy-preview-3155--design-insights.netlify.app/?path=/docs/components-pullrequests-prstateauthoricon--docs

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: components/PullRequests/PrStateAuthorIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    case state === "closed" && isMerged:
+      backgroundColor = "bg-purple-600";
+      break;
+  }
+
+  return (
+    <div className="relative w-max">
+      <Avatar contributor={author} size="medium" />
+      <div
+        className={`absolute -bottom-2 -right-2 p-1 rounded-full [&_svg]:absolute [&_svg]:top-[3.5px] [&_svg]:left-[3.5px] ${backgroundColor}`}
```
Comment (isabensusan) [inline components/PullRequests/PrStateAuthorIcon.tsx:56]: absolute nitpick on the style side, but I'd love to make some tiny changes to get here: 
image

move the icon circle a bit more to the bottom and to the right (had to hardcode values bc tailwind's values didn't work) and add a 2px white border to make it feel like it's separated from the avatar

` className={absolute -bottom-[-10px] -right-[-12px] p-1 border-[2px] border-white  rounded-full [&_svg]:absolute [&_svg]:top-[3.5px] [&_svg]:left-[3.5px] `

File: components/PullRequests/PrStateAuthorIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        style={{ width: "23px", height: "23px" }}
```
Comment (isabensusan) [inline components/PullRequests/PrStateAuthorIcon.tsx:57]: make the icon wrapper a bit bigger
`style={{ width: "25px", height: "25px" }}`

File: components/PullRequests/PrStateAuthorIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { GitMergeIcon, GitPullRequestClosedIcon, GitPullRequestIcon } from "@primer/octicons-react";
+import { Avatar } from "components/atoms/Avatar/avatar-hover-card";
+
+function getPullRequestStateIcon(
+  state: DbRepoPREvents["pr_state"],
+  isDraft: DbRepoPREvents["pr_is_draft"],
+  isMerged: DbRepoPREvents["pr_is_merged"]
+) {
+  const size = 16;
```
Comment (isabensusan) [inline components/PullRequests/PrStateAuthorIcon.tsx:9]: make it just a tad bit smaller to help the proportions

`const size = 14;`

File: components/PullRequests/PrStateAuthorIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  }
+}
+
+export const PrStateAuthorIcon = ({
+  state,
+  isDraft,
+  isMerged,
+  author,
+}: {
+  state: string;
```
Comment (brandonroberts) [inline components/PullRequests/PrStateAuthorIcon.tsx:29]: Restrict to valid states

```suggestion
  state: "open" | "closed";
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/PullRequests/PrStateAuthorIcon.tsx (hunk 4 above)
Comment (nickytonline) [reply components/PullRequests/PrStateAuthorIcon.tsx]: Good catch.

**Review by brandonroberts — APPROVED** [review]

**Review by jpmcb — APPROVED** [review]

**Review by isabensusan — APPROVED** [review]

**Pull request merged**